deptofdefense/move.mil
# [WIP] Implement secure_headers gem

## Checklist

I have…


## Summary of Changes

This pull request adds [Twitter's secure_headers gem](https://github.com/twitter/secureheaders) to the project and uses its features to implement more-secure HTTP response headers (e.g. `X-Frame-Options`, `Content-Security-Policy`) for routes served by the Rails app.

The move to do this was prompted by a conversation with @jobertabma at HackerOne. He noted the remote possibility of an attacker compromising a developer machine and taking advantage of our previous decision to offload header-setting to nginx instead of Rails/Action Dispatch.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. `git checkout secure-headers`,
1. set up development dependencies according to `CONTRIBUTING.md`,
1. run `bin/rails server`,
1. load up http://localhost:3000 and check your browser's console for the appropriate response headers.
